build/configs/rtl8720e/Make.defs

```diff
@@ -120,7 +120,12 @@ ifeq ($(CONFIG_FRAME_POINTER),y)
   ARCHOPTIMIZATION += -fomit-frame-pointer -mapcs -mno-sched-prolog
 endif
 
-ARCHCFLAGS = -fno-builtin -mcpu=cortex-m33 -mfpu=fpv5-sp-d16 -fno-common
+ARCHCFLAGS = -fno-builtin -mcpu=cortex-m33 -mfpu=fpv5-sp-d16
+
+ifneq ($(CONFIG_BUILD_FLAT),y)
+  ARCHCFLAGS += -fno-common
+  CFLAGS = -march=armv8-m.main+dsp
```

Comment:
I think single commit is enough and you can merge the commits. Can you check it please.